github/data_source_github_branch_test.go

```diff
@@ -0,0 +1,74 @@
+package github
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccGithubBranchDataSource_basic(t *testing.T) {
+
+	var (
+		name = "main"
+		repo = "test-repo"
+		rn   = "data.github_branch." + name
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckGithubBranchDataSourceConfig(name, repo, "master"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(rn, "repository", repo),
+					resource.TestCheckResourceAttr(rn, "branch", "master"),
+					resource.TestCheckResourceAttrSet(rn, "etag"),
+					resource.TestCheckResourceAttrSet(rn, "ref"),
+					resource.TestCheckResourceAttrSet(rn, "sha"),
+				),
+			},
+			{
+				Config: testAccCheckGithubBranchDataSourceConfig(name, repo, "master"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(rn, "repository", repo),
+					resource.TestCheckResourceAttr(rn, "branch", "master"),
+					resource.TestCheckResourceAttrSet(rn, "etag"),
+					resource.TestCheckResourceAttrSet(rn, "ref"),
+					resource.TestCheckResourceAttrSet(rn, "sha"),
+				),
+			},
```

Comment:
Is this a duplicate of the block above?

Reply:
@jcudit Yes. My understanding is that this checks for drift, essentially making sure my state doesn't accidentally drift between runs. This helped me catch an error with where I was setting `etag`.